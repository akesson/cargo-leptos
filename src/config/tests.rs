use camino::Utf8PathBuf;

use super::Config;

#[test]
fn test_project() {
    let cli = crate::Opts {
        release: false,
        project: None,
        verbose: 0,
    };

    let conf = Config::load(cli, &Utf8PathBuf::from("examples/project/Cargo.toml"), true).unwrap();

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace() {
    let cli = crate::Opts {
        release: false,
        project: None,
        verbose: 0,
    };

    let conf = Config::load(
        cli,
        &Utf8PathBuf::from("examples/workspace/Cargo.toml"),
        true,
    )
    .unwrap();

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace_project1() {
    let cli = crate::Opts {
        release: false,
        project: Some("project1".to_string()),
        verbose: 0,
    };

    let conf = Config::load(
        cli,
        &Utf8PathBuf::from("examples/workspace/Cargo.toml"),
        true,
    )
    .unwrap();

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace_project2() {
    let cli = crate::Opts {
        release: false,
        project: Some("project2".to_string()),
        verbose: 0,
    };

    let conf = Config::load(
        cli,
        &Utf8PathBuf::from("examples/workspace/Cargo.toml"),
        true,
    )
    .unwrap();

    insta::assert_debug_snapshot!(conf);
}
