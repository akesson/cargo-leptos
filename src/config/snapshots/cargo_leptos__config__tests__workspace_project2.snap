---
source: src/config/tests.rs
expression: conf
---
Config {
    projects: [
        Project {
            name: "project2",
            config: ProjectConfig {
                package_name: "project2",
                site_addr: 127.0.0.1:3000,
                site_root: "target/site",
                site_pkg_dir: "pkg",
                style_file: Some(
                    "src/main.scss",
                ),
                assets_dir: Some(
                    "src/assets",
                ),
                reload_port: 3001,
                end2end_cmd: None,
                end2end_dir: None,
                browserquery: "defaults",
                bin_target: None,
            },
            front_package: "project2 ..",
            front_profile: "debug",
            server_package: "project2 ..",
            server_target: "project2 (bin) ..",
            server_profile: "debug",
            watch: true,
            site: Site {
                file_reg: {},
                ext_file_reg: {},
            },
            paths: ProjectPaths {
                lib_crate_name: "project2",
                site_root: "target/site",
                site_pkg_dir: "target/site/pkg",
                front_dir: "proj2",
                server_dir: "proj2",
                target_dir: "target",
                wasm_file: SourcedSiteFile {
                    source: "target/front/wasm32-unknown-unknown/debug/project2",
                    dest: "target/site/pkg/project2.wasm",
                    site: "pkg/project2.wasm",
                },
                js_file: SiteFile {
                    dest: "target/site/pkg/project2.js",
                    site: "pkg/project2.js",
                },
                style_file: Some(
                    SourcedSiteFile {
                        source: "proj2/src/main.scss",
                        dest: "target/site/pkg/project2.css",
                        site: "pkg/project2.css",
                    },
                ),
                assets_dir: Some(
                    "proj2/src/assets",
                ),
                cargo_bin_file: "target/server/debug/project2",
                ..
            },
            ..
        },
    ],
    cli: Opts {
        release: false,
        project: Some(
            "project2",
        ),
        verbose: 0,
    },
    watch: true,
}
